ConnectionFactory.php

```diff
@@ -40,22 +43,23 @@ public function __construct(array $typesConfig)
      */
     public function createConnection(array $params, Configuration $config = null, EventManager $eventManager = null, array $mappingTypes = [])
     {
+        $connection = DriverManager::getConnection($params, $config, $eventManager);
+
         if (! $this->initialized) {
```

Comment:
there is an issue here now: registering the types in the registry only needs to be done once. But anything depending on the connection must be done again for each of them.
So making the code inside this `if (! $this->initialized) {` condition depend on the connection will not work fine.